### PR TITLE
implements ssl factory for tls 1.2 support

### PR DESCRIPTION
**Pull request by mustafacantekir**
implements ssl factory for tls 1.2 support

### CODE REVIEW

File: src/main/java/com/iyzipay/HttpClient.java [hunk 1]
```diff
@@ -20,6 +21,7 @@
     private static final String CONTENT_TYPE = "Content-Type";
     private static final String ACCEPT = "Accept";
     private static final int TIMEOUT = 140000;
+    private static final IyzipaySSLSocketFactory socketFactory = IyzipaySSLSocketFactory.getInstance();
```
Comment (AlicanAkkus) [inline src/main/java/com/iyzipay/HttpClient.java]: can we move this variable to head of class?

**Comment by AlicanAkkus** [comment]
conflict var, çözer misinizi? @mustafacantekir

File: src/main/java/com/iyzipay/IyzipaySSLSocketFactory.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    private void setTlsV12IfSupported() {
+        String[] supportedProtocols = new String[0];
+        try {
+            supportedProtocols = SSLContext.getDefault().getSupportedSSLParameters().getProtocols();
+        } catch (NoSuchAlgorithmException ignored) {
+        }
+
+        for (String protocol : supportedProtocols) {
```
Comment (ryumer) [inline src/main/java/com/iyzipay/IyzipaySSLSocketFactory.java]: strem.anyMatch olabilir mi burda?

Comment (mustafacantekir) [reply src/main/java/com/iyzipay/IyzipaySSLSocketFactory.java]: java 7 desteklemiyor.

File: src/main/java/com/iyzipay/HttpClient.java (hunk 1 above)
Comment (AlicanAkkus) [reply src/main/java/com/iyzipay/HttpClient.java]: variable name could be "iyzipaySSLSocketFactory"

**Pull request merged**